### PR TITLE
feat:Anonymize email data (RDCC-1748)

### PR DESCRIPTION
**Pull request by abhijitdiwan**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-1748

### Change description ###
feat:Anonymize email data (RDCC-1748)

**Does this PR introduce a breaking change?** (check one with "x")
No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/processor/TelemetryRequestProcessor.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    : pathVariable.indexOf("?"));
+
+                if (pathVariable.contains(".") && (pathVariable.contains("@"))) {
+                    values.add(pathVariable);
+                }
+
+                //anonymize  string values
+                for (String val : values) {
+                    String encodeVal = val.replace((val.substring(2, val.indexOf("."))), "****");
+                    requestTelemetry.setUrl(requestTelemetry.getUrlString().replace(val, encodeVal));
```
Comment (shashank-rastogi) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/processor/TelemetryRequestProcessor.java:60]: replace it till 2 characters before @

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/processor/TelemetryRequestProcessor.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                    && mp.getValue().contains(".")).map(NameValuePair::getValue).collect(toSet());
+
+                String pathVariable = getName(requestTelemetry.getUrlString());
+                pathVariable = pathVariable.substring(0, pathVariable.indexOf("?") == -1 ? pathVariable.length()
+                    : pathVariable.indexOf("?"));
+
+                if (pathVariable.contains(".") && (pathVariable.contains("@"))) {
+                    values.add(pathVariable);
+                }
+
```
Comment (shashank-rastogi) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/processor/TelemetryRequestProcessor.java:56]: do we need to do this for both params and path variables?

File: charts/rd-professional-api/values.preview.template.yaml [hunk 3]
```diff
@@ -15,5 +15,6 @@ java:
     rd:
       secrets:
         - s2s-secret
+        - AppInsightsInstrumentationKey
```
Comment (shreedhar-hmcts) [inline charts/rd-professional-api/values.preview.template.yaml:18]: Will this enable all PR level app insights in higher environment ? if it is then AAT app insights will show PR endpoints info.

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserController.java [hunk 4]
```diff
@@ -134,11 +134,7 @@ public ResponseEntity findUsersByOrganisation(@ApiParam(hidden = true) @OrgId St
                     @Authorization(value = "Authorization")
             }
     )
-    @ApiParam(
-            name = "email",
-            type = "string",
-            value = "The Email Address of the User to be retrieved"
-    )
+
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserController.java:137]: Check with BA,  if we can remove Email address description from swagger

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/processor/TelemetryRequestProcessor.java (hunk 2 above)
Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/processor/TelemetryRequestProcessor.java:56]: Yes.. we need both

File: charts/rd-professional-api/values.preview.template.yaml (hunk 3 above)
Comment (abhijitdiwan) [reply charts/rd-professional-api/values.preview.template.yaml:18]: yes it is for testing. I mnetioned in Abhijit Diwan:house_with_garden:  4:59 PM
https://github.com/hmcts/rd-professional-api/pull/704 have review on this ( note values-preview.yaml change temporary for testing)

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserController.java (hunk 4 above)
Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserController.java:137]: yes we can it is no use.

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/processor/TelemetryRequestProcessor.java (hunk 1 above)
Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/processor/TelemetryRequestProcessor.java:60]: in this domain will be visible and if & mil with less letters and can be again not secure.

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserController.java (hunk 4 above)
Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserController.java:137]: done

**Pull request merged**